crates/sui-types/src/gas_model/gas_v2.rs

```diff
@@ -380,6 +389,20 @@ impl<'a> SuiGasStatusAPI<'a> for SuiGasStatus<'a> {
         storage_cost
     }
 
+    fn track_deleted(&mut self, storage_rebate: u64) {
+        if self.is_unmetered() {
+            self.unmetered_storage_rebate += storage_rebate;
+        } else {
+            self.storage_rebate += storage_rebate;
+            self.deleted_rebate += storage_rebate;
+        }
+    }
+
+    fn track_for_out_of_gas(&mut self, storage_rebate: u64) {
```

Comment:
The one thing @lxfind was wondering about this when we spoke yesterday is whether this can make `computation_cost + storage_cost` > `budget` (and what the consequences of that are, if any).